R/list_of.R

```diff
@@ -159,13 +159,16 @@ vec_cast.vctrs_list_of <- function(x, to) {
 #' @export
 #' @method vec_cast.vctrs_list_of list
 vec_cast.vctrs_list_of.list <- function(x, to) {
+  # Casting list to list_of will warn/err if the cast is lossy,
+  # but the locations refer to the inner vectors,
+  # and the cast fails if all (vector) elements in a single (list) element
   as_list_of(x, .ptype = attr(to, "ptype"))
 }
 
 #' @export
 #' @method vec_cast.list vctrs_list_of
 vec_cast.list.vctrs_list_of <- function(x, to) {
-  warn_lossy_cast(x, to)
+  # Casting list_of to list is never lossy
```

Comment:
Isn't it? We're losing the type information about the contents, which seems lossy to me.

Reply:
Is this a lossy cast?

``` r
vctrs::vec_cast(1:3, list())
#> [[1]]
#> [1] 1
#> 
#> [[2]]
#> [1] 2
#> 
#> [[3]]
#> [1] 3
```

<sup>Created on 2019-01-25 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

I'd argue that a cast is lossy only if the data is changed, and an upcast (to a more general type) is never lossy.

Reply:
Oh got it.